lib/engine/step/buy_train.rb

```diff
@@ -53,6 +53,8 @@ def process_buy_train(action)
         buy_train_action(action)
         pass! unless can_buy_train?(action.entity)
       end
+
+      def swap_sell(_player, _corporation, _bundle, _pool_share); end
```

Comment:
why do you need swap_sell in buy_train?

Reply:
If I don't have it then I get an undefined method 'swap_sell' from Engine::Step::BuyTrain.
From MiniRacer.

I think that emergency funding requires the Sell shares part, and that uses swap functionality.
Probably 18MEX should implement swap sell (and share the code) as there might be a situation there you want to reduce your holdings in NdM to finance a train purchase. 
I guess I should subclass buy_train.rb for 18MEX and share the implementation for swap_sell. 

What do you think about that?

Reply:
maybe it's ok

Reply:
I think I will break out the swap_buy and swap_sell (and the helper function) to an 18Mex module, and import it in both 18Mex version of buy_sell_par_shares.rb and buy_train.rb. That way it will be available during emergency buy of trains in 18Mex. (The swap_buy wont be used there but it is good to have it in the same module.)

I'll add it as another commit to this PR.